python/global_search_completer/global_search_completer.py

```diff
@@ -0,0 +1,362 @@
+
+            return {
+                "type": data["type"],
+                "id": data["id"],
+                "name": data["name"],
+            }
+        else:
+            return None
+
+    def search(self, text):
+        """
+        Triggers the popup to display results based on the supplied text.
+
+        :param text: current contents of editor
+        """
+        if len(text) < 3:
+            # global search wont work with shorter than 3 len strings
+            # for these cases, clear the auto completer model fully
+            # there is no more work to do
+            self._clear_model(add_loading_item=False)
+            return
+
+        # now we are about to populate the model with data
+        # and therefore trigger the completer to pop up.
+        #
+        # The completer seems to have some internal properties
+        # which are transitory and won't last between sessions.
+        # for these, we have to set them up every time the
+        # completion process is about to start it seems.
+
+        # tell completer to render matches using our delegate
+        # configure how the popup should look
+
+        # deferred import to help documentation generation.
+        from .search_result_delegate import SearchResultDelegate
+
+        popup = self.popup()
+        self._delegate = SearchResultDelegate(popup)
+        popup.setItemDelegate(self._delegate)
+
+        # try to disconnect and reconnect the activated signal
+        # it seems this signal is lost every time the widget
+        # looses focus.
+        try:
+            self.activated[QtCore.QModelIndex].disconnect(self._on_select)
+        except:
+            self._bundle.log_debug(
+                "Could not disconnect global search activated signal prior to "
+                "reconnect. Looks like this connection must have been "
+                "discarded at some point along the way."
+            )
+
+        self.activated[QtCore.QModelIndex].connect(self._on_select)
+
+        # now clear the model
+        self._clear_model()
+
+        # clear download queue
+        if self.__sg_data_retriever:
+            self.__sg_data_retriever.clear()
+
+        # clear thumbnail map
+        self._thumb_map = {}
+
+        # kick off async data request from shotgun
+        # we request to run an arbitrary method in the worker thread
+        # this  _do_sg_global_search method will be called by the worker
+        # thread when the worker queue reaches that point and will
+        # call out to execute it. The data dictionary specified will
+        # be forwarded to the method.
+        data = {"text": text}
+        if self.__sg_data_retriever:
+            self._processing_id = self.__sg_data_retriever.execute_method(
+                self._do_sg_global_search, data)
+        else:
+            raise sgtk.TankError(
+                "Please associate this class with a background task processor.")
```

Comment:
Ah, perhaps not now that I've looked at set_bg_task_manager. Perhaps just replace "processor" with "manager"